libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -998,8 +1009,6 @@ void CompilerUtils::convertType(
 	default:
 		if (stackTypeCategory == Type::Category::Function && targetTypeCategory == Type::Category::Integer)
```

Comment:
I think this should be `if (stackTypeCategory == Type::Category::Function && targetTypeCategory == Type::Category::Address)`

Reply:
Currently the function recalls itself with integer types in the very beginning, if called with address types... that's a bit dirty, though, I'll clean it up.